common/achievement/bucket.spec.ts

```diff
@@ -0,0 +1,280 @@
+import moment from 'moment';
+import { bucketCreatorDefs } from './bucket';
+import { BucketCreatorContext, ContextMatch, ContextSubcourse, TimeBucket } from './types';
+
+describe('test create buckets by_lecture_start', () => {
+    // TODO: think about setting dow globally
+    moment.updateLocale('de', { week: { dow: 1 } });
+    jest.useFakeTimers().setSystemTime(new Date(2023, 7, 15));
+
+    const today = moment();
+    const yesterday = moment().subtract(1, 'day');
+    const lastWeek = moment().subtract(1, 'week');
+    const twoWeeksAgo = moment().subtract(2, 'week');
+
+    const tests: {
+        name: string;
+        expectedBuckets: TimeBucket[];
+        matches?: ContextMatch[];
+        subcourses?: ContextSubcourse[];
+    }[] = [
+        {
+            name: 'should create one bucket for a match with a single lecture',
+            expectedBuckets: [
+                {
+                    kind: 'time',
+                    startTime: moment('2023-08-14T21:50:00.000Z').toDate(),
+                    endTime: moment('2023-08-14T23:10:00.000Z').toDate(),
+                    relation: 'match',
+                },
+            ],
+            matches: [{ id: 1, relation: 'match', lecture: [{ start: today.toDate(), duration: 60 }] }],
+        },
+        {
+            name: 'should create multiple buckets for a match with multiple lectures',
```

Comment:
Huh, matches have lectures? Don't we call them appointments elsewhere?

Reply:
I don't know 😅 
I just followed the [prisma schema naming](https://github.com/corona-school/backend/blob/ceaf0f2a56ee769643c6948369862049dfd752f2/prisma/schema.prisma#L410).